app/gosns/common_errors.go

```diff
@@ -0,0 +1,34 @@
+package gosns
+
+import (
+	"net/http"
+
+	"github.com/Admiral-Piett/goaws/app"
+)
+
+var (
+	ErrNonExistentTopic = &app.SnsErrorType{
```

Comment:
This is a new point. In SQS codes, only few of errors are defined in `queue_attributes.go`.
To refer the error from implements, I tried to separate the error definitions from `init` in `gosns.go`.

If this way works, we can apply the same strategy also on gosqs codes.

Reply:
Hm, good start on this!   I'm not sure what the long term plan will be for the errors, since I have struggled with them too a bit.  This seems like a good plan for now, but lets put it in a file called `modes/errors.go`?